QUARKUS-4430.md

```diff
@@ -0,0 +1,35 @@
+# QUARKUS-3535 - Add a metric for rejected submissions to the pool in RHBQ 3.2
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-4430
+
+PR: https://github.com/quarkusio/quarkus/pull/39583
+
+Upstream issue: https://github.com/quarkusio/quarkus/issues/39581
+
+Verify that URIs in metrics tags are correctly recorded when they match a client or server pattern,
+especially in cases of redirection (302) and not found (404) responses.
+
+## Scope of the testing
+* Run Quarkus micrometer Prometheus application with OpenId Connect Keycloak and verify in case of redirections (302)
+  and not found (404) response the Uris metrics tags are correctly recorded.
+* The test coverage will be verified for JVM.
+
+
+## Existing test coverage
+* We have already tests related to monitoring metrics with Prometheus and OpenId Connect Keycloak but not specific test
+  about this improvement feature related to tag redirections and not found responses.
+ 
+
+### Impact on test suites and testing automation
+* New test class will be added into [monitoring](https://github.com/quarkus-qe/quarkus-test-suite/tree/main/monitoring/micrometer-prometheus-oidc) 
+
+### Impact on resources
+* One baremetal test execution in JVM mode
+* Test require any KeyCloak service container
```

Comment:
```suggestion
* Test require any Keycloak service container
```
I don't understand why it would require Keycloak at all, can you elaborate?

Reply:
Well, this issue development has been done for a Keycloak request, you can see here--> https://issues.redhat.com/browse/QUARKUS-4430 
_"Keycloak wants to raise a request to backport the enhancement for reflecting redirected URLs on HTTP Metrics. The Quarkus version in question is 3.8.

The relevant PR to be backported: https://github.com/quarkusio/quarkus/pull/39583 "_

Reply:
Let's add it to the motivation (the description paragraph), but you cannot use Keycloak in this testing because you don't have a use case. If you have in mind how to use the Keycloak in relation to the backport, please show me.

Reply:
For that reason, my idea was create the tests into https://github.com/quarkus-qe/quarkus-test-suite/blob/main/monitoring/micrometer-prometheus-oidc/ that it's used Keycloak and prometheus metric, so in that case we will have an integration test checking those urls for the cases Not found and redirection....

Reply:
I've moved the test just  to /monitoring/micrometer-prometheus without keycloak because is not relevant here.